and-bible/app/src/main/java/net/bible/service/sword/SwordEnvironmentInitialisation.java

```diff
@@ -59,11 +64,13 @@ public static void initialiseJSwordFolders() {
 				System.setProperty("jsword.home", moduleDir.getAbsolutePath());
 				CWProject.instance().setFrontendName("and-bible");
 
-				// the second value below is the one which is used in effectively all circumstances
-				CWProject.setHome("jsword.home", moduleDir.getAbsolutePath(), SharedConstants.MANUAL_INSTALL_DIR.getAbsolutePath());
-
-				// the following causes Sword to initialise itself and can take quite a few seconds
-				SwordBookPath.setAugmentPath(new File[] {SharedConstants.MANUAL_INSTALL_DIR});  // add manual install dir to this list
+				// Permission is requested at MainBibleActivitiy.checkSdcardReadPermission and app is restarted if permission is granted.
+				if(ContextCompat.checkSelfPermission(BibleApplication.getApplication(), Manifest.permission.READ_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED) {
+					// the second value below is the one which is used in effectively all circumstances
+					CWProject.setHome("jsword.home", moduleDir.getAbsolutePath(), SharedConstants.MANUAL_INSTALL_DIR.getAbsolutePath());
```

Comment:
Yes, but this line also includes moduleDir.getAbsolutePath(), which is the default install dir.  If this line is not executed (no manual install) then I suspect that the default install dir might also not be usable.

JSword config was setup by trial and error - I am no expert.  It would be worth commenting out this line to see if And Bible can still read and install modules.

Reply:
It appears that this was working properly without any change (I think it is because System.setProperty which is called on line 65. But I changed this anyway (added else block where setHome is called with altHomeDir = null).